icu4c/source/i18n/msgfmt.cpp

```diff
@@ -1673,6 +1673,11 @@ void MessageFormat::cacheExplicitFormats(UErrorCode& status) {
     }
 }
 
+int32_t MessageFormat::skipLeadingSpaces(UnicodeString& style) {
```

Comment:
Please add the following to class PatternProps (right before or after the existing pointer version) and use it here:
```
    /**
     * Skips over Pattern_White_Space starting at index start in s.
     * @return The smallest index at or after start with a non-white space character.
     */
    static int32_t skipWhiteSpace(const UnicodeString &s, int32_t start);
```

Reply:
Done